controllers/workflow_controller_test.go

```diff
@@ -0,0 +1,195 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dwsv1alpha1 "github.com/HewlettPackard/dws/api/v1alpha1"
+)
+
+var _ = Describe("Workflow Controller Test", func() {
+
+	var (
+		wf *dwsv1alpha1.Workflow
+	)
+
+	BeforeEach(func() {
+		wfid := uuid.NewString()[0:8]
+		wf = &dwsv1alpha1.Workflow{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      wfid,
+				Namespace: corev1.NamespaceDefault,
+			},
+			Spec: dwsv1alpha1.WorkflowSpec{
+				DesiredState: dwsv1alpha1.StateProposal,
+				WLMID:        "test",
+				JobID:        0,
+				UserID:       0,
+				GroupID:      0,
+				DWDirectives: []string{},
+			},
+			Status: dwsv1alpha1.WorkflowStatus{
```

Comment:
Leave Status empty when you're creating new resources.  This is filled in by the controllers.